### ISSUE TITLE
Invalid date in maintenance sensors

### ISSUE THREAD

**Issue by rpitera** [jeremywillans/ha-rest980-roomba#11]
Not sure if this is a bug or not, but I got no response in the forum thread so I'm asking here.

I struggled with getting this all to work, on one hand because I can't use packages due to the way I split my configs so I had to do everything manually and on the other hand, because I was doing this with the brand new i3 and at the time was under an NDA so I couldn't publicly ask for help.  Nonetheless I managed to get everything sorted for the most part and even have the map image working, though I still have a ways to go in getting it positioned properly.  

Thank you!  Your work is amazing and I love the information presented!

Now that the NDA has expired, I can finally ask for help with the one thing I can't seem to resolve - the maintenance sensors.   I have a working MQTT setup and discovery is working with the discovery_prefix option to the point where I was able to get to the step where you click on the check button card to create the sensors.

But mine show "Invalid Date" for all the maintenance sensor timestamp_friendly attributes and no data for the timestamp_timeout attribute in all but one sensor.  

As an example, here is the attributes for the sensor.vacuum_maint_clean_filter:

```
timestamp_friendly: Invalid Date
timeout: 10 days
timeout_timestamp: null
timeout_seconds: 864000
severity:
  - color: '#bf4060'
    value: '-10 days'
    seconds: -864000
  - color: '#bfb540'
    value: 0 days
    seconds: 0
friendly_name: vacuum_maint_clean_filter
icon: 'mdi:checkbox-marked'
device_class: timestamp
```

For the vacuum_maint_clean_wheel sensor however, I've got the following info:
```
timestamp: ''
timestamp_friendly: '12/31/1969, 7:00:00 PM'
timeout: 18 days
timeout_timestamp: 1555200
timeout_seconds: 1555200
severity:
  - color: '#bf4060'
    value: '-3 days'
    seconds: -259200
  - color: '#bfb540'
    value: 0 days
    seconds: 0
friendly_name: vacuum_maint_clean_wheel
icon: 'mdi:checkbox-marked'
device_class: timestamp
```
I'm not sure if this is because the i3 is new and needs a different approach or if the i3 simply doesn't keep track of maintenance or I have set something up incorrectly.   I've tried troubleshooting this myself looking at the output from http://my_ha_instance:3000/api/local/info/state with PostMan, looking at the traffic with MQTT explorer, etc. but I'm in over my head.

Can you suggest a troubleshooting approach or confirm that this won't work with the i3?  I got the impression that the maintenance sensor should work but wasn't sure if I set something up wrong or it just isn't supported.  For example, I know that (at the moment) room/region level cleaning isn't.

Thanks!

**Comment by rpitera** [comment]
I just clicked on my maintenance sensor check button cards - which were all reading "Over by 50 Years" and as I did so, they all started changing to proper/different dates.  I knew that I had to click on the each once the first time to set them up but apparently once set up they need to be clicked on again?   I'm going to leave this open for now until I can verify that these dates get updated and still work on my next vacuum run.

**Comment by jeremywillans** [comment]
Hey, sorry for the delay. The check button card sensors are independant of the vacuum and are just based around proposed timeline for maintance as suggested by iRobot - You wont find that these update based on your vacuum runs.

I know someone on the forum was going to look at integrating the two - but IMO its not worth it as its a 5 minute job to perfom weekly robot maintennce. 

Hope this helps.

**Comment by rpitera** [comment]
Hey, no worries - I imagine you're pretty busy!  I think I need to learn more about check button card; if I understood it better I would have realized how it worked.  Reading back on the maintenance automation should have been a clue - I kept looking for way that the sensors were updated that wasn't there.  If I understood better how the check buttons actually worked (and some of the use cases it was developed for, like feeding schedules for pets)  I might have realized that I not only needed to click on the boxes the FIRST time, but also a SECOND time once the topics were created to 'start the timer' so to speak.

I don't know if any additional documentation would be in order; while it would have helped me, looking at the issues and forum posts, it seemed like everyone else figured this out easily or already knew it.  Totally up to you if you decide there's any value in adding a sentence or two about this, but I do appreciate your response and REALLY appreciate your work!  So cool to have this integrated into HA.

Will leave this to you on whether you want respond or close this issue, but I'm good on my end.  Thanks again.
